editor/src/clj/editor/command_requests.clj

```diff
@@ -0,0 +1,270 @@
+;; Copyright 2020-2023 The Defold Foundation
+;; Copyright 2014-2020 King
+;; Copyright 2009-2014 Ragnar Svensson, Christian Murray
+;; Licensed under the Defold License version 1.0 (the "License"); you may not use
+;; this file except in compliance with the License.
+;;
+;; You may obtain a copy of the License, together with FAQs at
+;; https://www.defold.com/license
+;;
+;; Unless required by applicable law or agreed to in writing, software distributed
+;; under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+;; CONDITIONS OF ANY KIND, either express or implied. See the License for the
+;; specific language governing permissions and limitations under the License.
+
+(ns editor.command-requests
+  (:require [clojure.data.json :as json]
+            [clojure.string :as string]
+            [dynamo.graph :as g]
+            [editor.disk :as disk]
+            [editor.ui :as ui]
+            [service.log :as log])
+  (:import [java.nio.charset StandardCharsets]))
+
+(set! *warn-on-reflection* true)
+
+(defonce ^:const url-prefix "/command")
+
+(defonce ^:private supported-commands
+  ;; Notable exclusions:
+  ;; :save-all, :quit, anything that would open a modal dialog.
+  {:asset-portal
+   {:ui-handler :asset-portal
+    :help "Open the Asset Portal in a web browser."}
+
+   :async-reload
+   {:ui-handler :async-reload
+    :help "Reload all modified files from disk."}
+
+   :build
+   {:ui-handler :build
+    :help "Build and run the project."}
+
+   :build-html5
+   {:ui-handler :build-html5
+    :help "Build the project for HTML5 and open it in a web browser."}
+
+   :debugger-break
+   {:ui-handler :break
+    :help "Break into the debugger."}
+
+   :debugger-continue
+   {:ui-handler :continue
+    :help "Resume execution in the debugger."}
+
+   :debugger-detach
+   {:ui-handler :detach-debugger
+    :help "Detach the debugger from the running project."}
+
+   :debugger-start
+   {:ui-handler :start-debugger
+    :help "Start the project with the debugger attached."}
+
+   :debugger-step-into
+   {:ui-handler :step-into
+    :help "Step into the current expression in the debugger."}
+
+   :debugger-step-out
+   {:ui-handler :step-out
+    :help "Step out of the current expression in the debugger."}
+
+   :debugger-step-over
+   {:ui-handler :step-over
+    :help "Step over the current expression in the debugger."}
+
+   :debugger-stop
+   {:ui-handler :stop-debugger
+    :help "Stop the debugger and the running project."}
```

Comment:
I think commands such as these shouldn't trigger resource sync. On big projects, resource sync takes a while to run, so it will be annoying to unnecessarily wait e.g. 5 seconds to step over.

Reply:
That's a great point. I'll add it as metadata to the actions!